app/connectors/GuaranteeBalanceConnector.scala

```diff
@@ -91,7 +91,12 @@ class GuaranteeBalanceConnector @Inject() (http: HttpClient, appConfig: Frontend
               Right(BalanceRequestRateLimit)
             case Status.BAD_REQUEST =>
               logger.warn(s"[GuaranteeBalanceConnector][submitBalanceRequestV2] BAD_REQUEST response: ${response.body}")
-              Left(response)
+
+              if (response.body.toLowerCase.contains("the guarantee reference number is not in the correct format")) {
+                Right(BalanceRequestNotMatched(response.body))
```

Comment:
Is there maybe another way to identify this type of error. Might cause some problems if the message changes (maybe one for the API?)

Reply:
Arguably we should tighten up validation and prevent it from happening. But if we need to handle this we only have the error text to go on. An alternative would be to handle all 400 (BAD_REQUEST) this way and log the API error text? 

@MatthewAlexanderBeck

Reply:
Maybe thats an idea for a later refactor. Im wondering if we could get the API to return a code in the body rather than just raw text, that match its easier to match. Food for thought but I think the current approach should be fine for the time being. Logging the error text would also be a good suggestion.